### DISCUSSION TITLE
TypeError: Cannot read property 'readArmored' of undefined

### DISCUSSION THREAD

**Discussion by nils-kt** [openpgpjs/openpgpjs#1407] [Q&A]
Hello, I have a problem with my following code:

```js
import * as openpgp from "openpgp";

async function decrypt(_privateKey, message, result) {
  const { keys: [privateKey] } = await openpgp.key.readArmored(_privateKey);
  await privateKey.decrypt("password");
  const _message = await openpgp.readMessage({
    armoredMessage: message
  });
  const { data: decrypted } = await openpgp.decrypt({
    message: await openpgp.message.readArmored(_message),
    privateKeys: [privateKey]
  });
  console.log(decrypted);
  result(decrypted);
}
```

Everytime I get the following error: `UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'readArmored' of undefined`.

Arg message is like:
````
-----BEGIN PGP MESSAGE-----
Version: BCPG C# v1.8.9.0

hQGMAyEheGlIqF1+AQwAr/DBKqf6/HwCkboGsqwMy962OZou3KIBdiuGzM/fL0bq
FHc3csdWsT7WVHt7zGfRKU5SxrBmumSKIS/ebzEzuEJ1afshk11v+hkKIzLl708D
BjPeKLkJI+HlLCgm7dR1bvp4u9pd1mI9bDGfkbow4PtunGHSrqDKHKZSzEwkumVF
NoiayMu6GGSiO6yElQ2UOYiLAte1BOGI4FACsYdJ8m+Az5PEqwJLxwxve8EnHUQ5
m1jOgps3+l00YsuIVQ0lrciyxFV5ZrRyrhpzZRSAymG+icnutdMo8yRaiGK78hMk
Ui9+FUWAL9dNQp/J6vyIjwS3kG7ksegUHU1sKoJnN6QVXRRshznRF2DKT4HLoBcQ
CSq6bH0rf3PjpTbWw4d0fLIxt375k7f2ToX/ChUPUc7MzRqdFOb1l5XhXU79r3IS
0Hldq/8CeE0HEdxEdoRSLRkXwNMhGPnObK2YqbRy2lDZX2S6SE1jePxlVDUPQRTv
qVpeMy7vwjYx0C1mj2hg0kMBe2Qet35W3Y/yfWIdhA3DqpERCThfR/KsQxnf2WqL
Lo+cN0RbDm8wqvz8w78KgiVtI1Lvn/BxuVjD2PIc6Ac14eme
=4LX+
-----END PGP MESSAGE-----

**Comment by larabr** [comment]
Hi @nils-kt , which version of OpenPGP are you using? It looks like you might have the v5 version installed, but the code you are using is only compatible with v4. For examples on how to use v5, see the README: https://github.com/openpgpjs/openpgpjs#examples .

**Reply by nils-kt** [reply]
Now I get this error:
`Error decrypting message: Session key decryption failed.`

Maybe #1398

//Edit: Temporary fixed with allowInsecureDecryptionWithSigningKeys: true

**Reply by larabr** [reply]
If setting the config option fixed it, then the problem is with the primary key, not with the library, see https://github.com/openpgpjs/openpgpjs/issues/1126#issuecomment-683868831 . You'll have to set the config whenever you decrypt with that key.
